File: frontend/src/app/shared/service/admin/recomrecla/ReclamationEtudiantAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ReclamationEtudiantDto} from 'src/app/shared/model/recomrecla/ReclamationEtudiant.model';
import {ReclamationEtudiantCriteria} from 'src/app/shared/criteria/recomrecla/ReclamationEtudiantCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ReclamationEtudiantAdminService extends AbstractService<ReclamationEtudiantDto, ReclamationEtudiantCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ReclamationEtudiantDto {
        return new ReclamationEtudiantDto();
    }

    public constrcutCriteria(): ReclamationEtudiantCriteria {
        return new ReclamationEtudiantCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/reclamationEtudiant/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/recomrecla/ReclamationEtudiantAdminService.java
package ma.zs.alc.service.facade.admin.recomrecla;

import java.util.List;
import ma.zs.alc.bean.core.recomrecla.ReclamationEtudiant;
import ma.zs.alc.dao.criteria.core.recomrecla.ReclamationEtudiantCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ReclamationEtudiantAdminService extends  IService<ReclamationEtudiant,ReclamationEtudiantCriteria>  {

    List<ReclamationEtudiant> findByTypeReclamationEtudiantId(Long id);
    int deleteByTypeReclamationEtudiantId(Long id);
    long countByTypeReclamationEtudiantCode(String code);




}
